### DISCUSSION TITLE
Is it possible to run the command concurrently?

### DISCUSSION THREAD

**Discussion by andrewgurneyct** [deepjyoti30/ytmdl#268] [Q&A]
I am trying to see if I can spawn a few threads running at the same time and I have run into a few errors. The most recent are:

- Failed while moving with error: list index out of range
- Error occurred:  [Errno 2] No such file or directory: '/home/adg/.cache/ytmdl/I#Wonder.mp3' . Exiting. Pass `--ignore-errors` if you want errorslike this to be ignored

I wonder if there would be a way to specify the temp directory or otherwise so that I can get multiple runs going at the same time.

**Comment by deepjyoti30** [comment]
Based on the details you have provided, I think the moving part is failing due to `ytmdl` removing un recognized files from the temp directory before moving the actual file. This means that whichever thread gets to the moving step first will remove the files that other threads are using and all the other threads will fail.

You can change the temp dir through the config but that would mean requiring one config for each thread.

Can you perhaps, share more context on why you want to run `ytmdl` in multiple threads? There might be some alternative way to do what you're trying to achieve instead of running multiple threads.
